ArthurAtangana/SYSC3303A_Project
# Implementing request/reply protocol in messaging

## What type of PR is this? (check all applicable)

- [ ] 🐛 Bug Fix (maybe? wasn't really a bug...)

## Description
- Transmitter/Receivers now go through a receive reply loop to catch dropped packets hopefully

## How to test
- Run UDP mains, look to make sure system doesn't hang. 
- A performance difference between UDP/DMA should be helpful to see how much the communication is slowing the system down.

## Added/updated tests?

- [ ] Yes: Just a scenario test and a config for it.
- [X] No, and this is why: Testable through UDP mains

## Screenshots

## Notes
